test/utils/poll-helper.ts

```diff
@@ -0,0 +1,131 @@
+import { expect } from "chai";
+import fetchMock from "fetch-mock";
+import * as sinon from "sinon";
+
+import { HTTPMethod } from "../../src/api/RestAPI.js";
+import { POLLING_INTERVAL, POLLING_TIMEOUT } from "../../src/common/constants.js";
+import { TimeoutError } from "../../src/errors/TimeoutError.js";
+
+const tickPoll = async (sandbox: sinon.SinonSandbox, ticks = 3) => {
```

Comment:
`for await`, use async iterators here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of